test/e2e/apps/statefulset.go

```diff
@@ -1156,6 +1157,45 @@ var _ = SIGDescribe("StatefulSet", func() {
 				return false, nil
 			})
 		})
+
+		/*
+			Release: v1.32
+			Testname: StatefulSet, PodIndexLabel
+			Description: When Pods for a StatefulSet is created it MUST be created with PodIndexLabel feature
+			Attempt to read the PodIndexLabel key and value
+		*/
+		framework.It("pods for Statefulset should have PodIndexLabel with right index", feature.PodIndexLabel, func(ctx context.Context) {
```

Comment:
I'm inclined to push the label check into this test: https://github.com/kubernetes/kubernetes/blob/d9b95ea94f41b942440adbdfdbe35fb78a2ed6f7/test/e2e/apps/statefulset.go#L144-L187
which checks basic statefulset functionality.